numpy/random/src/distributions/distributions.c

```diff
@@ -1337,6 +1348,14 @@ uint64_t random_bounded_uint64(bitgen_t *bitgen_state, uint64_t off,
     return off;
   } else if (rng <= 0xFFFFFFFFUL) {
     /* Call 32-bit generator if range in 32-bit. */
+    if (rng == 0xFFFFFFFFUL) {
+      /*
+       * The 32-bit Lemire method does not handle rng=0xFFFFFFFF, so we'll
+       * call next_uint32 directly.  This also works when use_masked is True,
+       * so we handle both cases here.
+       */
+      return off + (uint64_t) next_uint32(bitgen_state);
```

Comment:
I think the same would apply here as well.

Reply:
Yes, this needs a test.  When calling `randint` with scalar values of `low` and `high`, this path is not reached, even when `size` is `None`.    It turns out that the function `random_bounded_uint64` is called by the code that handles broadcasting for `random.randint`.  I'll add a repeatability test for that.